failing build eskipfile TestWatchUpdateRecover

The build of #593 is failing on travis (maybe flaky test, because yesterday it worked without code change in this).

```--- FAIL: TestWatchUpdateRecover (0.04s)
	watch_test.go:93: failed to load route for: /foo
FAIL
FAIL	github.com/zalando/skipper/eskipfile	0.237s
```

--- a/eskipfile/watch_test.go
+++ b/eskipfile/watch_test.go
@@ -1,6 +1,7 @@
 package eskipfile
 
 import (
+	"io/ioutil"
 	"net/http"
 	"net/url"
 	"os"
@@ -44,15 +45,20 @@ func deleteFile(t *testing.T) {
 }
 
 func createFileWith(content string, t *testing.T) {
-	f, err := os.Create(testWatchFile)
+	tmpfile, err := ioutil.TempFile("", "watch_test-*")
 	if err != nil {
 		t.Fatal(err)
 	}
-	defer f.Close()
+	defer os.Remove(tmpfile.Name())
 
-	_, еrr := f.WriteString(content)
-	if еrr != nil {
-		t.Fatal(еrr)
+	_, err = tmpfile.WriteString(content)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = os.Rename(tmpfile.Name(), testWatchFile)
+	if err != nil {
+		t.Fatal(err)
 	}
 }
 
@@ -86,53 +92,40 @@ func initWatchTest(t *testing.T) *watchTest {
 
 func (t *watchTest) testFail(path string) {
 	if r, _ := t.routing.Route(&http.Request{URL: &url.URL{Path: path}}); r != nil {
-		t.testing.Error("unexpected route received for:", path)
 		t.testing.Log("got:     ", r.Id)
 		t.testing.Log("expected: nil")
-		t.testing.FailNow()
+		t.testing.Fatalf("unexpected route received for: %v", path)
 	}
 }
 
 func (t *watchTest) testSuccess(id, path, backend string) {
 	r, _ := t.routing.Route(&http.Request{URL: &url.URL{Path: path}})
 	if r == nil {
-		t.testing.Error("failed to load route for:", path)
-		t.testing.FailNow()
+		t.testing.Fatalf("failed to load route for: %v", path)
 		return
 	}
 
 	if r.Id != id || r.Backend != backend {
-		t.testing.Error("unexpected route received")
 		t.testing.Log("got:     ", r.Id, backend)
 		t.testing.Log("expected:", id, r.Backend)
-		t.testing.FailNow()
+		t.testing.Fatal("unexpected route received")
 	}
 }
 
-func (t *watchTest) timeoutOrFailInitial() {
-	if t.testing.Failed() {
-		return
-	}
-
+func (t *watchTest) timeoutInitial() {
 	defer t.log.Reset()
-	if err := t.log.WaitFor("route settings applied", 90*time.Millisecond); err != nil {
-		// timeout is also good, the routing handles its own
-		return
+	err := t.log.WaitFor("route settings applied", 90*time.Millisecond)
+	if err == nil {
+		t.testing.Fatal("expected timeout, got route settings applied")
+	} else if err != loggingtest.ErrWaitTimeout {
+		t.testing.Fatalf("unexpected error: %v", err)
 	}
-
-	t.testFail("/foo")
-	t.testFail("/bar")
-	t.testFail("/baz")
 }
 
 func (t *watchTest) timeoutAndSucceedInitial() {
-	if t.testing.Failed() {
-		return
-	}
-
 	defer t.log.Reset()
 	if err := t.log.WaitFor("route settings applied", 90*time.Millisecond); err == nil {
-		t.testing.Error("unexpected change detected")
+		t.testing.Fatal("unexpected change detected")
 	}
 
 	t.testSuccess("foo", "/foo", "https://foo.example.org")
@@ -141,10 +134,6 @@ func (t *watchTest) timeoutAndSucceedInitial() {
 }
 
 func (t *watchTest) waitAndFailInitial() {
-	if t.testing.Failed() {
-		return
-	}
-
 	defer t.log.Reset()
 	if err := t.log.WaitFor("route settings applied", 90*time.Millisecond); err != nil {
 		t.testing.Fatal(err)
@@ -156,10 +145,6 @@ func (t *watchTest) waitAndFailInitial() {
 }
 
 func (t *watchTest) waitAndSucceedInitial() {
-	if t.testing.Failed() {
-		return
-	}
-
 	defer t.log.Reset()
 	if err := t.log.WaitFor("route settings applied", 90*time.Millisecond); err != nil {
 		t.testing.Fatal(err)
@@ -171,10 +156,6 @@ func (t *watchTest) waitAndSucceedInitial() {
 }
 
 func (t *watchTest) waitAndSucceedUpdated() {
-	if t.testing.Failed() {
-		return
-	}
-
 	defer t.log.Reset()
 	if err := t.log.WaitFor("route settings applied", 90*time.Millisecond); err != nil {
 		t.testing.Fatal(err)
@@ -194,13 +175,13 @@ func (t *watchTest) close() {
 func TestWatchInitialFails(t *testing.T) {
 	test := initWatchTest(t)
 	defer test.close()
-	test.timeoutOrFailInitial()
+	test.timeoutInitial()
 }
 
 func TestWatchInitialRecovers(t *testing.T) {
 	test := initWatchTest(t)
 	defer test.close()
-	test.timeoutOrFailInitial()
+	test.timeoutInitial()
 	createFile(t)
 	defer deleteFile(t)
 	test.waitAndSucceedInitial()

--- a/logging/loggingtest/logger.go
+++ b/logging/loggingtest/logger.go
@@ -123,10 +123,7 @@ func New() *Logger {
 				lw.count(req)
 			case c := <-clear:
 				lw.clear()
-				select {
-				case c <- struct{}{}:
-				default:
-				}
+				c <- struct{}{}
 			case m := <-mute:
 				muted = m
 			case <-quit:

